modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/AbstractCacheDumpTest.java

```diff
@@ -0,0 +1,465 @@
+
+            assertEquals(GRP, ccfg0.getGroupName());
+            assertEquals(CACHE_0, ccfg0.getName());
+
+            assertEquals(GRP, ccfg1.getGroupName());
+            assertEquals(CACHE_1, ccfg1.getName());
+
+            assertFalse(ccfgs.get(0).sql());
+            assertFalse(ccfgs.get(1).sql());
+            assertTrue(ccfgs.get(0).queryEntities().isEmpty());
+            assertTrue(ccfgs.get(1).queryEntities().isEmpty());
+
+            List<Integer> parts = dump.partitions(nodeDir, CU.cacheId(DEFAULT_CACHE_NAME));
+
+            for (int part : parts) {
+                try (DumpedPartitionIterator iter = dump.iterator(nodeDir, CU.cacheId(DEFAULT_CACHE_NAME), part)) {
+                    while (iter.hasNext()) {
+                        DumpEntry e = iter.next();
+
+                        checkDefaultCacheEntry(e, coCtx);
+
+                        keys.add(e.key().<Integer>value(coCtx, true));
+
+                        dfltDumpSz++;
+                    }
+                }
+            }
+
+            parts = dump.partitions(nodeDir, CU.cacheId(GRP));
+
+            for (int part : parts) {
+                try (DumpedPartitionIterator iter = dump.iterator(nodeDir, CU.cacheId(GRP), part)) {
+                    while (iter.hasNext()) {
+                        DumpEntry e = iter.next();
+
+                        checkGroupEntry(e, coCtx0, coCtx1);
+
+                        grpDumpSz++;
+                    }
+                }
+            }
+        }
+
+        assertEquals(KEYS_CNT + KEYS_CNT * backups, dfltDumpSz);
+        assertEquals(2 * (KEYS_CNT + KEYS_CNT * backups), grpDumpSz);
+
+        IntStream.range(0, KEYS_CNT).forEach(key -> assertTrue(keys.contains(key)));
+    }
+
+    /** */
+    protected void checkDefaultCacheEntry(DumpEntry e, CacheObjectContext coCtx) {
+        assertNotNull(e);
+
+        Integer key = e.key().<Integer>value(coCtx, true);
+
+        assertEquals(key, e.value().<Integer>value(coCtx, true));
+    }
+
+    /** */
+    protected void checkGroupEntry(DumpEntry e, CacheObjectContext coCtx0, CacheObjectContext coCtx1) {
+        assertNotNull(e);
+
+        if (e.cacheId() == CU.cacheId(CACHE_0))
+            assertEquals(USER_FACTORY.apply(e.key().value(coCtx0, true)), e.value().value(coCtx0, true));
+        else {
+            assertNotNull(e.key().<Key>value(coCtx1, true));
+            assertNotNull(e.value().<Value>value(coCtx1, true));
+        }
+    }
+
+    /** */
+    protected void insertOrUpdate(IgniteEx ignite, int i) {
+        insertOrUpdate(ignite, i, i);
+    }
+
+    /** */
+    protected void insertOrUpdate(IgniteEx ignite, int i, int val) {
+        ignite.cache(DEFAULT_CACHE_NAME).put(i, val);
+        IgniteCache<Object, Object> cache = ignite.cache(CACHE_0);
+        IgniteCache<Object, Object> cache1 = ignite.cache(CACHE_1);
+
+        if (mode == CacheAtomicityMode.TRANSACTIONAL) {
+            try (Transaction tx = ignite.transactions().txStart()) {
```

Comment:
No need explicitly start transaction for single put. It automatically starts for every single put/delete/etc operations.